### PR TITLE
bda.plone.shop issue #67 Sort the user filter, reorder filter contents.

### PR DESCRIPTION
**Pull request by pcdummy**
This is the @@orders fix for [bda.p.shop #67](https://github.com/bluedynamics/bda.plone.shop/issues/67)

Signed-off-by: Rene Jochum <rene@jochums.at>

### CODE REVIEW

File: src/bda/plone/orders/vocabularies.py [hunk 1]
```diff
@@ -96,15 +97,25 @@ def customers_vocab_for(user=None):
...
+        name = None
         if customer:
             # soft dep on bda.plone.shop
             first = customer.getProperty('firstname', '')
             last = customer.getProperty('lastname', '')
+            email = customer.getProperty('email', '')
             # fallback
             full = customer.getProperty('fullname', '')
-            name = (first or last) and '{0}, {1}'.format(first, last) or full
+            name = (first or last) and '{0}, {1}'.format(last, first) or full
```
Comment (thet) [inline src/bda/plone/orders/vocabularies.py]: plz instead:
``            name = u'{0}, {1}'.format(last, first) if (last or first) else full``

File: src/bda/plone/orders/vocabularies.py [hunk 2]
```diff
@@ -96,15 +97,25 @@ def customers_vocab_for(user=None):
...
+
+        if email and name:
+            title = '{0} ({1}) - {2}'.format(name, creator, email)
```
Comment (thet) [inline src/bda/plone/orders/vocabularies.py]: plz. add unicode string literal ``title = u'{0} ({1}) - {2}'.format(name, creator, email)``

File: src/bda/plone/orders/browser/bookings.py [hunk 3]
```diff
@@ -166,20 +167,20 @@ def render_buyable_uid(self, colname, record):
...
-            + ': ' + str(bookings_quantity) + '</span>' \
-            '<span>' +\
+            + u': ' + safe_unicode(bookings_quantity) + u'</span>' \
+            u'<span>' +\
             translate(
                 _("bookings_total_sum", default=u"Bookings total sum"),
                 self.request
             ) \
-            + ': ' + str(bookings_total_sum) + '</td></tr>'
+            + u': ' + bookings_total_sum + u'</td></tr>'
```
Comment (thet) [inline src/bda/plone/orders/browser/bookings.py]: maybe refactor that to make use multi line strings and the format string method?
```u"""blabla""".format(...)```

File: src/bda/plone/orders/vocabularies.py [hunk 4]
```diff
@@ -96,15 +97,25 @@ def customers_vocab_for(user=None):
...
+        name = None
         if customer:
             # soft dep on bda.plone.shop
             first = customer.getProperty('firstname', '')
             last = customer.getProperty('lastname', '')
+            email = customer.getProperty('email', '')
             # fallback
             full = customer.getProperty('fullname', '')
-            name = (first or last) and '{0}, {1}'.format(first, last) or full
+            name = (first or last) and u'{0}, {1}'.format(last, first) or full
```
Comment (thet) [inline src/bda/plone/orders/vocabularies.py]: We should also change from the deprecated form of conditional clauses using boolean expression to ternary form.
Instead:
``(first or last) and u'{0}, {1}'.format(last, first) or full``
Better:
``u'{0}, {1}'.format(last, first) if (first or last) else full``

There are some cases where boolean form won't evaluate correctly.

Comment (pcdummy) [inline src/bda/plone/orders/vocabularies.py]: Fixed in the next commit, waiting for one from @rnixx

**Comment by thet** [comment]
LGTM

**Pull request merged**